src/transformers/pipelines/base.py

```diff
@@ -869,6 +869,8 @@ def _ensure_tensor_on_device(self, inputs, device):
         elif isinstance(inputs, tuple):
             return tuple([self._ensure_tensor_on_device(item, device) for item in inputs])
         elif isinstance(inputs, torch.Tensor):
+            if device == torch.device("cpu") and inputs.dtype in {torch.float16, torch.bfloat16}:
+                inputs = inputs.float()
             return inputs.to(device)
```

Comment:
won't it be faster to copy to cpu first and then upcast? since copy then needs to copy only half the data